src/lib.rs

```diff
@@ -2,4 +2,9 @@
 #![allow(non_camel_case_types)]
 #![allow(non_snake_case)]
 
-include!(concat!(env!("OUT_DIR"), "/bindings.rs"));
+mod gen {
```

Comment:
Can you explain a bit these modifications ?

Reply:
Hi @Wenzel  and I'm very sorry for the delay.

Your question is indeed reasonable. In putting `bindings.rs` into a private module `gen`, I do not want the upper level crate to use `pub` (functions and structs) generated by `bindgen` directly, that is because they (i.e. functions and structs) are too low-level (they are not Rust at all), and worse, they're sometimes unnecessary.

For example, in `bindings.rs`  there are some exports:

```rust
pub type __uint8_t = ::std::os::raw::c_uchar;
pub type __int16_t = ::std::os::raw::c_short;
pub type __uint16_t = ::std::os::raw::c_ushort;
pub type __int32_t = ::std::os::raw::c_int;
pub type __uint32_t = ::std::os::raw::c_uint;
pub type __int64_t = ::std::os::raw::c_long;
pub type __uint64_t = ::std::os::raw::c_ulong;
```

which are useless: they are not used anywhere but because some "mysterious" limits of `bindgen` we cannot blacklist them if we want to keep `Rust` unions from `C` unions. And because they are automatically generated, it's easy to conflict with the same types of other low-level bindings  (which may be also auto generated by bindgen) of other `C` libraries.

Another example is the following auto generated export

```rust
extern "C" {
    pub fn xc_domain_send_trigger(
        xch: *mut xc_interface,
        domid: u32,
        trigger: u32,
        vcpu: u32,
    ) -> ::std::os::raw::c_int;
}
```

Such a function is too "low-level": `xch` is a raw pointer but it's must be non-null (unfortunately, `bindgen` didn't know about that), so it's actually a `std::ptr::NonNull<xc_interface>`, or even better a `&mut xc_interface`. So there is a better export:

```rust
extern "C" {
    pub fn xc_domain_send_trigger(
        xch: &mut xc_interface,
        domid: u32,
        trigger: u32,
        vcpu: u32,
    ) -> i32
}
```

IMHO, a low-level `Rust` binding (i.e. a `-sys` crate), while giving an ABI-compatible interface to `C` library (if not, the binding is simply unusable), it should make the interface more rusty (by using idioms of `Rust`).

Actually, I want to re-export completely all auto-generated results of `bindgen`, but doing this by hand is a time-consuming task, so currently I simply put:

```rust
pub use gen::*;
```

But I will make it more concrete in future (if you're agree).

Reply:
We can optimize this in the future and rewrite the auto generated functions by more Rusty and manually defined ones.

I'm testing your PR to check if it compiles